files/autopause/autopause-daemon.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+exec 1>/tmp/terminal-mc
+
+. /autopause/autopause-fcns.sh
+
+sudo /usr/sbin/knockd -c /autopause/knockd-config.cfg -d
+if [ $? -ne 0 ] ; then
+  while :
+  do
+    if [[ -n $(ps -o comm | grep java) ]] ; then
+      break
+    fi
+    sleep 0.1
+  done
+  echo "[Autopause loop] Failed to start knockd daemon."
+  echo "[Autopause loop] Possible cause: docker's host network mode."
+  echo "[Autopause loop] Recreate without host mode or disable autopause functionality."
+  echo "[Autopause loop] Stopping server."
+  killall -SIGTERM java
+  exit 1
+fi
+
+STATE=K
+TIME_THRESH=$(($(current_uptime)+$AUTOPAUSE_TIMEOUT_KN))
+
+while :
+do
+  case X$STATE in
+  XK)
+    # Knocked
+    if java_clients_connected ; then
+      echo "[Autopause loop] Client connected - waiting for disconnect"
+      STATE=E
+    else
+      if [[ $(current_uptime) -ge $TIME_THRESH ]] ; then
+        echo "[Autopause loop] No client connected since startup - stopping"
+        /autopause/pause.sh
```

Comment:
What about check for a specific file before enabling auto pause, like level.dat or something, should only be there when map is finish generated?

Reply:
Hmm, parsing the log in the data directory is possible, but we'd have to parse for something in every version. `Level.dat` always exists except for worldgen. The port is being listened to since the log entry, so probably after mod initialization, but definitely before world generation / loading. We could do something like 10min or 4*Knocked, whichever is higher, after port listening. I kind of want to avoid another environment variable and testing the versions. The knocked TO is mainly for the server being queried but not connected to.

Reply:
I can only say for vanilla
[Server thread/INFO]: Done
and forge
[Server thread/INFO]: Done

Reply:
@eihns I was thinking more in the Minecraft versions, e.g. Vanilla 1.3 only prints:
```
        at net.minecraft.server.i.run(SourceFile:539)
2020-05-19 12:24:13 [INFO] Done (3.315s)! For help, type "help" or "?"
2020-05-19 12:24:13 [INFO] Starting remote control listener
2020-05-19 12:24:13 [INFO] RCON running on 0.0.0.0:25575
[Autopause loop] No client connected since startup - stopping
```
which would fail your match, whereas my forge 1.7.10 prints (when searching for Done):
```
[12:36:14] [Server thread/INFO]: Done.
[12:36:24][FINE][noppes.npcs.controllers.LinkedNpcController:63] Done loading Linked Npcs
[12:36:30] [Server thread/INFO]: Done (6.053s)! For help, type "help" or "?"
[12:36:31][FINE][noppes.npcs.controllers.DialogController:34] Done loading Dialogs
```

Reply:
Dont know what you mean, both print some sort of done when finish? *done*?

Reply:
I opted for the new environment variable, specifying TO after the server starts listening